include/transform_output/code_optimize.hpp

```diff
@@ -0,0 +1,175 @@
+
+/**
+* This function is used to set the input and output file paths
+* for all the four components of the simulator. The idea behind
+* creating this function is to remove redundancies in project
+* and utilize common code functionlilty. This will help in reducing
+* total LOC. It takes module_name as input and set the i/p and o/p
+* variables to respective file paths.
+* @param module_name of type string
+*/
+
+void set_file_name(std::string module_name){
+
+    if(module_name == "subnet"){
+       
+        //#define input_file_name "test/data/subnet_test_output.txt"
+        //#define output_file_name "test/data/subnet_test_output_transform.txt"
+        input_file_name = "test/data/subnet_test_output.txt";
+        output_file_name = "test/data/subnet_test_output_transform.txt";
+        printf("subnet files assigned %s \t %s\n",input_file_name,output_file_name);
+        
+
+    }else if(module_name == "receiver"){
+       
+        //#define input_file_name "test/data/receiver_test_output.txt"
+        //#define output_file_name "test/data/receiver_test_output_transform.txt"
+        input_file_name = "test/data/receiver_test_output.txt";
+        output_file_name = "test/data/receiver_test_output_transform.txt";
+        printf("receiver files assigned %s \t %s\n",input_file_name,output_file_name);  
+
+    }else if(module_name == "sender"){
+       
+        //#define input_file_name "test/data/sender_test_output.txt"
+        //#define output_file_name "test/data/sender_test_output_transform.txt"
+        input_file_name = "test/data/sender_test_output.txt";
+        output_file_name = "test/data/sender_test_output_transform.txt";
+        printf("sender files assigned %s \t %s\n",input_file_name,output_file_name);  
+
+    }else if(module_name == "abp_simulator"){
+       
+        //#define input_file_name "data/top_model/abp_output.txt"
+        //#define output_file_name "data/top_model/abp_output_transform.txt"
+        input_file_name = "data/top_model/abp_output.txt";
+        output_file_name = "data/top_model/abp_output_transform.txt";
+        printf("abp_simulator files assigned %s \t %s\n",input_file_name,output_file_name);  
+
+    }
+
+}
+
+
+/**
+ * To generate messages and operation logs which are being passed
+ * during execution time of this application and storing them
+ * in the receiver_test_output file as indicated out_data.
+ */
+static std::ofstream out_data(input_file_name);
+  
+
+/**
+ * This is a common sink provider structure 
+ * which calls the ostream that is the output stream
+ * and it return data stored in the file
+ */
+struct oss_sink_provider{
+    static std::ostream& sink() {          
+    	return out_data;
+    }
+};
+
+/**
+ * This following section of code is redundant in all four
+ * components and un-used. So we commented out this section.
+ * Please do not remove it as it may serve as a reference in futire.
+ * Its a good coding practice to keep the old code commented as 

```

Comment:
It is not a good practice to keep old code commented. Deprecated code should be removed.
That is why you use github and proper name for the commits. If you need old code, you go back to the commit where it was removed and you take it.